### PR TITLE
Nightly clippy suggestions

### PR DESCRIPTION
**Pull request by kirawi**

### CODE REVIEW

**Comment by kirawi** [comment]
I'll fix the submodule changes in a bit.

File: helix-view/src/graphics.rs [hunk 1]
```diff
@@ -185,15 +194,15 @@ impl Rect {
         let y1 = max(self.y, other.y);
         let x2 = min(self.x + self.width, other.x + other.width);
         let y2 = min(self.y + self.height, other.y + other.height);
-        Rect {
+        Self {
```
Comment (pickfire) [inline helix-view/src/graphics.rs:197]: Is this a clippy suggestion?

File: helix-view/src/graphics.rs [hunk 2]
```diff
@@ -170,7 +178,8 @@ impl Rect {
     }
 
     /// Calculate the intersection between two [`Rect`]s.
-    pub fn intersection(self, other: Rect) -> Rect {
+    #[must_use]
+    pub fn intersection(self, other: Self) -> Self {
```
Comment (pickfire) [inline helix-view/src/graphics.rs:182]: Is this a clippy suggestion? I personally prefer the older version because it's clearer.

Comment (archseer) [reply helix-view/src/graphics.rs:182]: I prefer Self but this looks like an optional clippy lint that's off by default

**Comment by archseer** [comment]
Let's tackle these once they trickle down to stable. Various lints tend to be overly noisy in nightly and get tuned by the time they arrive on stable.

**Pull request closed without merging**